src/mutate.rs

```diff
@@ -102,8 +118,92 @@ impl Client {
             .await?
             .into_inner();
 
-        self.collection_cache.update_timestamp(&collection_name, result.timestamp);
+        self.collection_cache
+            .update_timestamp(&collection_name, result.timestamp);
+
+        Ok(result)
+    }
+
+    pub async fn delete(
+        &self,
+        collection_name: impl Into<String>,
+        options: &DeleteOptions,
+    ) -> Result<crate::proto::milvus::MutationResult> {
+        if options.ids.len() > 0 && options.filter.len() > 0 {
+            return Err(Error::InvalidParameter(
+                "len(filter)".to_owned(),
+                options.filter.len().to_string(),
+            ));
+        }
+
+        if options.ids.len() == 0 && options.filter.len() == 0 {
+            return Err(Error::InvalidParameter(
+                "len(filter)".to_owned(),
+                options.filter.len().to_string(),
+            ));
+        }
+
+        let collection_name = collection_name.into();
+
+        let mut expr = match options.filter.len() {
+            0 => {
+                let collection = self.collection_cache.get(&collection_name).await?;
```

Comment:
what would happen if another sdk is modifing this collection in another process?